sdk/src/main/scala/io/horizen/history/AbstractHistory.scala

```diff
@@ -321,13 +320,44 @@ abstract class AbstractHistory[
     indexes :+ 1
   }
 
+  // calculate the other node approximate height knowing that the sequence was created using the method knownBlocksHeightToSync
+  private def calculateOtherNodeApproxHeight(startingHeight: Int, sequenceSize: Int): Int = {
+    if (startingHeight == 1) {
+      -1 // it is not possible to calculate the other node best block approximate height in this case
+    } else {
+      var currentHeight = startingHeight
+      var currentSequencePosition = sequenceSize - 1
+      var step = 1 // step used in case the input sequence size is > 10
+      while (currentSequencePosition > 0) {
+        if (currentSequencePosition < 10) {
+          currentHeight += 1
+        } else {
+          step *= 2
+          currentHeight += step
+        }
+        currentSequencePosition -= 1
+      }
+      currentHeight
+    }
+  }
+
+  // check the presence of a block on the current history at the second divergent suffix sequence height
```

Comment:
let's change the comment and the name of the method to be more clear
checkForkAtSecondToLastDivergentSuffixIndex
or something similar :)

Reply:
method name update, the one proposed seems fine and it was used!